sympy/ntheory/residue_ntheory.py

```diff
@@ -834,6 +833,7 @@ def nthroot_mod(a, n, p, all_roots=False):
     >>> nthroot_mod(68, 3, 109)
     23
     """
+    a = a % p
```

Comment:
this should go after the import and a test like `assert nthroot_mod(Dummy(odd=True), 3, 2) == 1` could be added.

Reply:
Done. One doubt,  is it okay to put `a=a%p` before `a=as_int(a)`. Because when it was after `a=as_int(a)` the dummy variable case was giving an error.

Reply:
yes